### PR TITLE
Add/update documentation on the dashboard flows

### PR DESCRIPTION
**Pull request by andrewballantyne**
Update to docs to showcase the new format and to rework some of our existing documentations (renames & reorganizations). This is likely the first of many steps to get docs in the right spot.

Of the over 2k updated lines, 1.5k reside in the source file for the images used to craft this PR `docs/process-definition/meta/imagesSource.drawio`.

You can view the full doc changes on my fork: https://github.com/andrewballantyne/odh-dashboard/blob/flow-documentation/README.md -- everything _should_ be navigable. 

The new flow:

fullFlow

The branch strategy:

branchesComplex

### CODE REVIEW

File: docs/dev-setup.md
Comment (pnaik1) [inline docs/dev-setup.md:63]: We neither have a  script for storybook nor test:storybook

Comment (andrewballantyne) [reply docs/dev-setup.md:63]: @Gkrumbach07 @christianvogt I think when you guys changed the landscape you didn't update the docs.

Could you guys spin that off in another issue?

File: docs/process-definition/branches.md
Comment (manaswinidas) [inline docs/process-definition/branches.md:63]: Is the apostrophe after `within` intentional?

File: docs/release-steps.md
Comment (manaswinidas) [inline docs/release-steps.md:54]: Apostrophe after `within`.

File: docs/architecture.md
Comment (manaswinidas) [inline docs/architecture.md:116]: Apostrophe after `within`

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: docs/dev-setup.md
Comment (Gkrumbach07) [reply docs/dev-setup.md:63]: I think it was updated in a different doc but not this one

File: docs/process-definition/flow-feature.md
Comment (manaswinidas) [inline docs/process-definition/flow-feature.md:78]: I didn't catch this earlier, the last sentence sounds a bit incomplete. `if it is not an Advisor...` Does this mean, `if there is no UI lead, an Advisor....`?

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: docs/dev-setup.md
Comment (Gkrumbach07) [reply docs/dev-setup.md:63]: here is where the up to date details are: https://github.com/opendatahub-io/odh-dashboard/blob/main/CONTRIBUTING.md#debugging-and-testing

im guessing you got that text from the readme

Comment (andrewballantyne) [reply docs/dev-setup.md:63]: `dev-setup.md` is entirely what was on the root README, so the mistake exists on `main` today

File: docs/process-definition/flow-feature.md
Comment (andrewballantyne) [reply docs/process-definition/flow-feature.md:78]: I changed it a bit more lol, because I still can't ready it seems.

>   - UI Leads do not have to be [Advisors], but if it is not an Advisor, there will be one that shadows to assist with the architecture

Comment (manaswinidas) [reply docs/process-definition/flow-feature.md]: How about `but in the absence of an Advisor...`?

Comment (manaswinidas) [reply docs/process-definition/flow-feature.md]: Yes, this makes sense.

File: docs/process-definition/incubation.md
Comment (lucferbux) [inline docs/process-definition/incubation.md:21]: I know we've discussed this but I'm still convinced that SMEs/future leads should be updating feature branches during incubation time periodically (could be once a sprint or smth). The reason behind is to detect conflicts in main. This conflicts are gonna be present when we merge main into incubation too since incubation is a superset of the other feature branches, but if we just fix incubation, the conflict will be there when merging the feature branch back to main, that's why I think that should be a good practice to follow.

File: docs/dev-setup.md
Comment (andrewballantyne) [reply docs/dev-setup.md:19]: Sounds good -- this was previous docs, `dev-setup.md` was on our root README

File: docs/release-steps.md
Comment (andrewballantyne) [reply docs/release-steps.md:38]: This can be done separately... this documents the flow today and not looking to solve CI gaps. Feel free to log something and address it concurrently. There will be a need to update the docs more later and we can fix this when we have the CI in place.

File: docs/release-steps.md
Comment (andrewballantyne) [reply docs/release-steps.md:74]: Ah, dang, I kept editing and I did not see this. Good catch.

File: docs/process-definition/incubation.md
Comment (andrewballantyne) [reply docs/process-definition/incubation.md:21]: It won't stop conflicts, but it could be a good idea to do that. I'm not sure how we'd enforce that... perhaps it's a task for leads and we setup a slack reminder or something. I'll not mention it in this pass of the docs, we can discuss internally and figure out the best way to go about that.

File: docs/process-definition/README.md
Comment (alexcreasy) [inline docs/process-definition/README.md:21]: Nit: should probably read:

"note: There is a lot of detail in here, this is all the flows together. See each flow individually below.

Note the "in" instead of "is"

File: docs/process-definition/README.md
Comment (alexcreasy) [inline docs/process-definition/README.md:29]: nit: grammatically clunky.

Suggestion: "There are also some additional resources that may be helpful:"

File: docs/process-definition/advisors.md
Comment (alexcreasy) [inline docs/process-definition/advisors.md:12]: I'm not entirely sure what this question is asking?

File: docs/process-definition/branches.md
Comment (alexcreasy) [inline docs/process-definition/branches.md:20]: "Fork" shouldn't be capitalised here.

File: docs/process-definition/branches.md
Comment (alexcreasy) [inline docs/process-definition/branches.md:46]: nit: last sentence is clunky, it could perhaps read:

"Conflicts are normal *when* merging into it."

File: docs/process-definition/incubation.md
Comment (alexcreasy) [reply docs/process-definition/incubation.md:21]: Yeah, I think this is a good practice, but I don't think we can get away from having to solve the conflicts multiple times. This should at least reduce the likelihood of being held up by multiple conflicts at critical periods, so still a really good idea.

File: docs/process-definition/meta/branchesComplex.png
Comment (alexcreasy) [inline docs/process-definition/meta/branchesComplex.png:1]: I like this diagram, to me it makes sense especially showing that main can be merged into incubation at multiple points. I just hope this isn't intuitive to me because I've banged my head against the process a few times, hopefully we've got enough eyes on this that someone will speak up if it doesn't make sense :)

File: docs/process-definition/releases.md
Comment (alexcreasy) [inline docs/process-definition/releases.md:11]: nit: second sentence should probably begin:
"It ends with the git tags"

File: docs/user-interaction.md
Comment (alexcreasy) [inline docs/user-interaction.md:5]: Oxford comma alert.... ;)

(Joking, just thought this is probably among the most pedantic reviews I've ever done, so may as well lean all the way into it :D)

**Review by alexcreasy — COMMENTED** [review]
Almost everything I've flagged is nit picking over the grammar, but there is one point I raised where I genuinely didn't understand what you were trying to say (around the responsibilities of approvers) - I'd address that one, but consider the others optional ;)

**Comment by dpanshug** [comment]
In the CONTRIBUTING.md [file](https://github.com/andrewballantyne/odh-dashboard/blob/73b410c70c109864b0db9b68c2c26813b1f48b1b/CONTRIBUTING.md?plain=1#L7):

Original:
```
You can check the requirements of ODH in the [README section](./README.md#requirements).
``` 

Suggested Improvement:

```suggestion
To review the ODH requirements, please refer to the [dev setup documentation](./docs/dev-setup.md#requirements).
```

File: docs/process-definition/advisors.md
Comment (andrewballantyne) [reply docs/process-definition/advisors.md:12]: Improved wording and more clarity on the roles.

**Comment by andrewballantyne** [comment]
Okay, going to merge this PR now.

I welcome continued content reviews, I also have a series of ToDo items I need to swing back with as well. Holding this up serves nothing and hurts nothing if we merge it now. It'll be a continued improvement process to get through all this anyways.

Since this is not code, I'll just manually add both labels. We'll iterate as needed on any confusion or content issues.

Thanks all who gave reviews 👍

**Pull request merged**
